core/src/main/scala/org/http4s/parser/AdditionalRules.scala

```diff
@@ -28,4 +28,11 @@ private[http4s] object AdditionalRules {
       case _: NumberFormatException => None
     }
   }
+
+  val Long: P[Long] = Rfc7230.token.mapFilter { s =>
+    try Some(s.toLong)
+    catch {
+      case _: NumberFormatException => None
+    }
+  }
```

Comment:
Is this going to accept negative values?  If you start from `digits` instead of `token`, you guard against that, and then the try/catch takes care of (unlikely) overflows.

Reply:
Oh, I see.  You're parsing everything as a long, and then running it through the `fromLong` constructor.  That'll work.